proxy/reverse.go

```diff
@@ -55,6 +57,19 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 	proxyreq := new(http.Request)
 	*proxyreq = *clientreq
 
+	var proxybody []byte
```

Comment:
Around the bytes.Buffer? Is there an existing `ResetClose` in the etcd codebase that I could re-use? Did some quick grep'ing but I didn't see anything.

If you mean around the clientreq's `Body` it would still require us to buffer the request someplace. This was the simplest method to do that.